### PR TITLE
Implement out of context stack unwinder

### PR DESCRIPTION
**Pull request by mikem8361**
Decode the eh frame info found in the in-memory module image
and pass it back to the remote libunwind8 to do the unwind.

Added an all managed threads option -all the "clrstack" (sos ClrStack).

### CODE REVIEW

**Comment by mikem8361** [comment]
@dotnet-bot test Tizen armel Cross Release Build

**Comment by mikem8361** [comment]
I addressed this code review feedback.  There is a new squashed commit.  Thanks.

File: src/ToolBox/SOS/Strike/strike.cpp [hunk 1]
```diff
@@ -331,14 +331,22 @@ DEBUG_STACK_FRAME g_Frames[MAX_STACK_FRAMES];
...
+GetContextStackTrace(ULONG osThreadId, PULONG pnumFrames)
 {
     PDEBUG_CONTROL4 debugControl4;
     HRESULT hr;
 
     // Do we have advanced capability?
     if ((hr = g_ExtControl->QueryInterface(__uuidof(IDebugControl4), (void **)&debugControl4)) == S_OK)
     {
+        ULONG oldId, id;
+        g_ExtSystem->GetCurrentThreadId(&oldId);
```
Comment (janvorli) [inline src/ToolBox/SOS/Strike/strike.cpp]: Can you please explain why we need do to this dance with swapping the thread id?

**Comment by mikem8361** [comment]
GetContextStackTrace needed to work on the specific thread instead of just the current one to support the new “clrstack -all” option. Even though IDebugControl4::GetContextStackTrace can take a starting thread context, the Linux implementation does support it. It was easier to just change the current thread context and back to the original. Updating the lldb plugin is hard because of the various versions of lldb people out there are using.

**Comment by mikem8361** [comment]
@dotnet-bot test Tizen armel Cross Release Build

**Comment by mikem8361** [comment]
@dotnet-bot test Tizen armel Cross Debug Build

**Pull request merged**